tests/functional/ProjectTask.php

```diff
@@ -402,39 +402,49 @@ public function testCloneProjectTask()
         $this->string($clonedSubtask2->fields['name'])->isEqualTo($subtask2->fields['name'] . ' (copy)');
     }
 
-    public function testAutoSetDateForAdd()
+    public function dataTestAdd()
     {
-        $_SESSION['glpi_currenttime'] = '2023-10-10 10:10:10';
-        // Create a project
         $project = $this->createItem('Project', [
             'name' => 'Project 1',
         ]);
+        return [
+            [
+                [ // Task with percent_done < 100 and > 0
+                    'projects_id' => $project->getID(),
+                    'name'        => 'Task 1',
+                    'percent_done' => '5',
+                ],
+            ],
+            [
+                [ // Task with percent_done = 100
+                    'projects_id' => $project->getID(),
+                    'name'        => 'Task 2',
+                    'percent_done' => '100',
+                ],
+            ],
+        ];
+    }
 
-        // Create a project task with percent_done < 100 and > 0
-        $task = $this->createItem('ProjectTask', [
-            'projects_id' => $project->getID(),
-            'name'        => 'Task 1',
-            'percent_done' => '5',
-        ]);
-
-        // Check if the task has been added with the current date
-        $this->string($task->fields['real_start_date'])->isEqualTo(\Session::getCurrentTime());
+    /**
+     * @dataprovider dataTestAdd
+     */
+    public function testAutoSetDateForAdd($input)
+    {
+        $_SESSION['glpi_currenttime'] = '2023-10-10 10:10:10';
+        // Create a project
 
-        // Create a project task with percent_done = 100
-        $task = $this->createItem('ProjectTask', [
-            'projects_id' => $project->getID(),
-            'name'        => 'Task 2',
-            'percent_done' => '100',
-        ]);
+        // Create a project task with percent_done < 100 and > 0
+        $task = $this->createItem('ProjectTask', $input);
 
         // Check if the task has been added with the current date
         $this->string($task->fields['real_start_date'])->isEqualTo(\Session::getCurrentTime());
-        $this->string($task->fields['real_end_date'])->isEqualTo(\Session::getCurrentTime());
+        if ($input['percent_done'] == 100) {
+            $this->string($task->fields['real_end_date'])->isEqualTo(\Session::getCurrentTime());
+        }
     }
 
-    public function testAutoSetDateForUpdate()
+    public function dataTestUpdate()
```

Comment:
Some cases are not tested :
 - update of project task already have a `real_start_date` and/or a `real_end_date`, with or without a `real_start_date` and/or a `real_end_date` in input,
 - a project task updated without  a `percent_done` input
 - a project task updated with an empty `percent_done` input
 - a project task updated with empty strings in `real_start_date` and/or `real_end_date`
 - a project task updated with null values in `real_start_date` and/or `real_end_date`